Foundation/NSURLSession/NativeProtocol.swift

```diff
@@ -0,0 +1,700 @@
+// Foundation/NSURLSession/TransferState.swift - NSURLSession & libcurl
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+// -----------------------------------------------------------------------------
+///
+/// The state of a single transfer.
+/// These are libcurl helpers for the URLSession API code.
+/// - SeeAlso: https://curl.haxx.se/libcurl/c/
+/// - SeeAlso: NSURLSession.swift
+///
+// -----------------------------------------------------------------------------
+
+import CoreFoundation
+import Dispatch
+
+internal let enableLibcurlDebugOutput: Bool = {
+    return (ProcessInfo.processInfo.environment["URLSessionDebugLibcurl"] != nil)
+}()
+internal let enableDebugOutput: Bool = {
+    return (ProcessInfo.processInfo.environment["URLSessionDebug"] != nil)
+}()
+
```

Comment:
It may help if we could add a line or two describing the role of this new internal class :)